src/adaptation/massmatrix.jl

```diff
@@ -153,6 +176,74 @@ function get_estimation(wv::WelfordVar{T}) where {T<:AbstractFloat}
     return n / ((n + 5) * (n - 1)) * M .+ ϵ * (5 / (n + 5))
 end
 
+## Nutpie-style diagonal mass matrix estimator (using positions and gradients)
+
+mutable struct NutpieVar{T<:AbstractFloat,E<:AbstractVecOrMat{T},V<:AbstractVecOrMat{T}} <: DiagMatrixEstimator{T}
```

Comment:
Does this have to be `mutable`?

Reply:
See e.g. [here](https://github.com/TuringLang/AdvancedHMC.jl/blob/dc8dc1cab540fbd3224bc934bf46d27bf0e48d4b/src/adaptation/massmatrix.jl#L57-L59), which implies among other things the presence of a (mutable) `n` field.